homeassistant/components/ibeacon/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for iBeacon Tracker integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import bluetooth
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+)
+
+from .const import CONF_MIN_RSSI, DEFAULT_MIN_RSSI, DOMAIN
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for iBeacon Tracker."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if not bluetooth.async_scanner_count(self.hass, connectable=False):
+            return self.async_abort(reason="bluetooth_not_available")
+
+        if user_input is not None:
+            return self.async_create_entry(title="iBeacon Tracker", data={})
+
+        return self.async_show_form(step_id="user")
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for iBeacons."""
+
+    def __init__(self, entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.entry = entry
+
+    async def async_step_init(self, user_input: dict | None = None) -> FlowResult:
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_MIN_RSSI,
```

Comment:
I don't think we should have these kinds of options. The user shouldn't be able or need to tune the noise floor. If this is a problem, I don't think this is a good way to integrate this device type.

Reply:
I added this because users complain about being able to see their neighbors' devices, but maybe this is a bit premature as I don't think iBeacons are common enough that might actually matter and they could just disable the device if they don't want it.  My neighbor only has one that I can see so maybe we can just rip this out.

Will do some more testing at other locations

Reply:
An alternative would be to just ask users to disable all the devices it doesn't want to track. Thoughts @bdraco ?

Reply:
Did we move or remove the option to disable devices in the UI?  I can't seem to find it on any of my installs

Reply:
https://github.com/home-assistant/core/pull/78843